src/Raven.Server/ServerWide/Commands/IncrementClusterIdentitiesBatchCommand.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using Raven.Client.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Voron;
+using Voron.Data.Tables;
+
+namespace Raven.Server.ServerWide.Commands
+{
+    public class IncrementClusterIdentitiesBatchCommand : UpdateValueForDatabaseCommand
+    {
+        public List<string> Identities;
+
+    
+        public IncrementClusterIdentitiesBatchCommand()
+            : base(null)
+        {
+            // for deserialization
+        }
+
+
+        public IncrementClusterIdentitiesBatchCommand(string databaseName, List<string> identities):base(databaseName)
+        {
+            Identities = identities;
+            DatabaseName = databaseName;
+        }
+     
+        public override void Execute(TransactionOperationContext context, Table items, long index, DatabaseRecord record, bool isPassive, out object result)
+        {
+            var identitiesTree = context.Transaction.InnerTransaction.ReadTree(ClusterStateMachine.Identities);
+            var results = new LinkedList<long>();
```

Comment:
Use a `List<long>`, it is easier to manage.

Reply:
The linkedList was there in order not to pay O(n) on each removal (RemoveFirst vs. RemoveAt(0)).  I did https://github.com/ravendb/ravendb/pull/4074 and therefor it can be List again